rust/windows-client/src-tauri/src/gui.rs

```diff
@@ -0,0 +1,210 @@
+            });
+
+            // From https://github.com/FabianLars/tauri-plugin-deep-link/blob/main/example/main.rs
+            let handle = app.handle();
+            tauri_plugin_deep_link::register("firezone", move |request| {
+                dbg!(&request);
+                handle.trigger_global("scheme-request-received", Some(request));
+            })?;
+            Ok(())
+        })
+        .build(tauri::generate_context!())?
+        .run(|_app_handle, event| {
+            if let tauri::RunEvent::ExitRequested { api, .. } = event {
+                // Don't exit if we close our main window
+                // https://tauri.app/v1/guides/features/system-tray/#preventing-the-app-from-closing
+
+                api.prevent_exit();
+            }
+        });
+    Ok(())
+}
+
+#[tauri::command]
+async fn apply_account_id(account_id: &str) -> StdResult<(), String> {
+    // These functions are wrapped since `anyhow::Error` doesn't implement `serde::Serialize` and so it can't go back to JS
+    apply_account_id_inner(account_id)
+        .await
+        .map_err(|e| format!("{e}"))
+}
+
+#[tauri::command]
+async fn apply_advanced_settings(
+    auth_base_url: &str,
+    api_url: &str,
+    log_filter: &str,
+) -> StdResult<(), String> {
+    apply_advanced_settings_inner(auth_base_url, api_url, log_filter)
+        .await
+        .map_err(|e| format!("{e}"))
+}
+
+async fn apply_account_id_inner(account_id: &str) -> Result<()> {
+    let dirs = crate::cli::get_project_dirs()?;
+    let dir = dirs.config_local_dir();
+    tokio::fs::create_dir_all(dir).await?;
+    let path = dir.join("account_id.json");
+
+    let j = serde_json::json!({
+        "account_id": account_id,
+    });
+
+    // TODO: Do something more robust. This will corrupt the file if we lose power while writing. Not the end of the world, but annoying.
+    tokio::fs::write(path, j.to_string()).await?;
+
+    // TODO: This sleep is just for testing, remove it before it ships
+    tokio::time::sleep(Duration::from_secs(2)).await;
+    Ok(())
+}
+
+async fn apply_advanced_settings_inner(
+    auth_base_url: &str,
+    api_url: &str,
+    log_filter: &str,
+) -> Result<()> {
+    let dirs = crate::cli::get_project_dirs()?;
+    let dir = dirs.config_local_dir();
+    tokio::fs::create_dir_all(dir).await?;
+    let path = dir.join("advanced_settings.json");
+
+    let j = serde_json::json!({
+        "auth_base_url": auth_base_url,
+        "api_url": api_url,
+        "log_filter": log_filter,
+    });
+
+    // TODO: Do something more robust. This will corrupt the file if we lose power while writing. Not the end of the world, but annoying.
+    tokio::fs::write(path, j.to_string()).await?;
+
+    // TODO: This sleep is just for testing, remove it before it ships
+    tokio::time::sleep(Duration::from_secs(2)).await;
+    Ok(())
+}
+
+fn signed_in_menu(user_email: &str, resources: &[(&str, &str)]) -> SystemTrayMenu {
```

Comment:
let's use an struct for `resources` instead of tuples.

Reply:
Sure, added it in the new branch